node/external/timemachine/fee/feeComputer.go

```diff
@@ -34,14 +29,11 @@ func NewFeeComputer(args ArgsNewFeeComputer) (*feeComputer, error) {
 	computer := &feeComputer{
 		builtInFunctionsCostHandler: args.BuiltInFunctionsCostHandler,
 		economicsConfig:             args.EconomicsConfig,
-		// TODO: use a LRU cache instead
-		economicsInstances: make(map[uint32]economicsDataWithComputeFee),
-		enableEpochsConfig: args.EnableEpochsConfig,
-		txVersionChecker:   args.TxVersionChecker,
+		enableEpochsHandler:         args.EnableEpochsHandler,
+		txVersionChecker:            args.TxVersionChecker,
 	}
 
-	// Create some economics data instance (but do not save them) in order to validate the arguments:
-	_, err = computer.createEconomicsInstance(0)
+	computer.economicsInstance, err = computer.createEconomicsInstance()
```

Comment:
I don't think you still need to create an economics instance, you would receive the real one as argument.
also no need for an enable epochs handler as well (you only use it to create the economics instance)

Reply:
updated